plugin/storage/cassandra/factory.go

```diff
@@ -131,6 +137,262 @@ func (f *Factory) configureFromOptions(o *Options) {
+			replication_factor, err = strconv.Atoi(replication_factor_string)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		replication = fmt.Sprintf("{'class': 'NetworkTopologyStrategy', '%s': '%v' }", datacenter, replication_factor)
+	}
+
+	trace_ttl_string := os.Getenv(TRACE_TTL)
+	trace_ttl := 172800
+	if trace_ttl_string != "" {
+		trace_ttl, err = strconv.Atoi(trace_ttl_string)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	dependencies_ttl_string := os.Getenv(DEPENDENCIES_TTL)
+	dependencies_ttl := 0
+	if dependencies_ttl_string != "" {
+		dependencies_ttl, err = strconv.Atoi(dependencies_ttl_string)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cas_version_string := os.Getenv(VERSION)
+	cas_version := 4
+	if cas_version_string != "" {
+		cas_version, err = strconv.Atoi(cas_version_string)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	keyspace := os.Getenv(KEYSPACE)
+	if keyspace == "" {
+		keyspace = fmt.Sprint("jaeger_v1_%s", datacenter)
+	}
+
+	var isMatch bool
+	isMatch, err = regexp.MatchString("[^a-zA-Z0-9_]", keyspace)
+	if err != nil {
+		return nil, err
+	}
+
+	if isMatch {
+		return nil, fmt.Errorf(`invalid characters in KEYSPACE=%s parameter, please use letters, digits or underscores`, keyspace)
+	}
+
+	if compaction_window := os.Getenv(COMPACTION_WINDOW); compaction_window != `` {
+		isMatch, err = regexp.MatchString("^[0-9]+[mhd]$", compaction_window)
+		if err != nil {
+			return nil, err
+		}
+
+		if !isMatch {
+			return nil, fmt.Errorf("Invalid compaction window size format. Please use numeric value followed by 'm' for minutes, 'h' for hours, or 'd' for days.")
+		}
+
+		compaction_window_size, err = strconv.Atoi(compaction_window[:len(compaction_window)-1])
+		if err != nil {
+			return nil, err
+		}
+
+		compaction_window_unit = compaction_window[len(compaction_window)-1:]
+	} else {
+		trace_ttl_minutes := trace_ttl / 60
+
+		compaction_window_size = (trace_ttl_minutes + 30 - 1) / 30
+		compaction_window_unit = "m"
+	}
+
+	switch compaction_window_unit {
+	case `m`:
+		compaction_window_unit = `MINUTES`
+	case `h`:
+		compaction_window_unit = `HOURS`
+	case `d`:
+		compaction_window_unit = `DAYS`
+	}
+
+	return &StorageConfigParams{
+		mode:                   mode,
+		datacenter:             datacenter,
+		trace_ttl:              trace_ttl,
+		dependencies_ttl:       dependencies_ttl,
+		keyspace:               keyspace,
+		replication_factor:     replication_factor,
+		replication:            replication,
+		cas_version:            cas_version,
+		compaction_window_size: compaction_window_size,
+		compaction_window_unit: compaction_window_unit,
+	}, nil
+}
+
+// Embed all the template files in binaries
+
+//go:embed schema/v001.cql.tmpl
+//go:embed schema/v002.cql.tmpl
+//go:embed schema/v003.cql.tmpl
+//go:embed schema/v004.cql.tmpl
+var schemaFile embed.FS
+
+func handleTemplateReplacements(data []byte, params *StorageConfigParams) []byte {
+	templateKeysValuePairs := map[string]string{
```

Comment:
It's better to rewrite the schema with Go template syntax and use that, because then the data binding can be directly against the config struct (better type safety).

Reply:
Done. Thanks